# Medical Cost Analysis
In this page of the notebook, I have imported medical cost data from Kaggle (https://www.kaggle.com/datasets/nanditapore/medical-cost-dataset?resource=download) and preform the first steps of exploratory data analysis.

In [8]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import seaborn as sns

In [9]:
def configure_plots():
    '''Configures plots by making some quality of life adjustments'''
    for _ in range(2):
        plt.rcParams['figure.figsize'] = [16, 9]
        plt.rcParams['axes.titlesize'] = 20
        plt.rcParams['axes.labelsize'] = 16
        plt.rcParams['xtick.labelsize'] = 14
        plt.rcParams['ytick.labelsize'] = 14
        plt.rcParams['lines.linewidth'] = 2

In [10]:
%%javascript
IPython.OutputArea.prototype._should_scroll = function(lines) {
    return false;
}

<IPython.core.display.Javascript object>

In [11]:
data = pd.read_csv("./archive (2)/medical_cost.csv")

In [16]:
data.head()

Id  age     sex     bmi  children smoker     region      charges
0   1   19  female  27.900         0    yes  southwest  16884.92400
1   2   18    male  33.770         1     no  southeast   1725.55230
2   3   28    male  33.000         3     no  southeast   4449.46200
3   4   33    male  22.705         0     no  northwest  21984.47061
4   5   32    male  28.880         0     no  northwest   3866.85520

In [17]:
data.shape

(1338, 8)

In [18]:
data.columns

Index(['Id', 'age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'], dtype='object')

##### Inital Notes About Data
The categories included in the dataset are ID (which replaces the patients name), age, sex, bmi (body mass index), children (the number of children the patient has), smoker, region (seperated by northeast, northwest, southeast, and southwest), and charges (the medical charges incurred by the patient). 
There are 1338 data points in the set before processing.

In [23]:
#cleaning the data (numerical)
clean_bmi = data['bmi'].replace(' ', '_').replace("(", "").replace(")", "")
clean_charges = data['charges'].replace(' ', '_').replace("(", "").replace(")", "").round(2)

Things to explore: age and cost, sex and cost, age and sex, region and cost, smoker and age and sex and cost, bmi and cost, children and bmi,